engine/runtime/src/main/java/org/enso/interpreter/runtime/data/Array.java

```diff
@@ -128,6 +131,54 @@ boolean hasType() {
     return true;
   }
 
+  @ExplodeLoop
+  private boolean hasWarningElements(Object[] items) {
+    for (int i = 0; i < items.length; i++) {
+      if (items[i] instanceof WithWarnings) {
+        return true;
+      }
+    }
+    return false;
+  }
+
+  @ExportMessage
+  boolean hasWarnings() {
+    if (withWarnings == null) {
+      withWarnings = hasWarningElements(items);
```

Comment:
How it can be `null` when the same `hasWarningsElements` function is invoke in the constructor?

Reply:
Survived some refactoring. Fixed